### PR TITLE
Use identity service to get auth tokens

### PR DESCRIPTION
**Pull request by liamaharon**

### CODE REVIEW

**Comment by liamaharon** [comment]
Needs https://github.com/hubiinetwork/nahmii-sdk/issues/17 to be resolved

File: src/containers/HubiiApiHoc/saga.js [hunk 1]
```diff
@@ -98,28 +97,57 @@ export function* loadTransactions({ address }, endpoint) {
   }
 }
 
+export function* requestToken() {
+  while (true) { // eslint-disable-line no-constant-condition
+    try {
+      const network = yield select(makeSelectCurrentNetwork());
+      const nahmiiProvider = new nahmii.NahmiiProvider(
+        network.walletApiEndpoint.slice(7, -1),
```
Comment (katat) [inline src/containers/HubiiApiHoc/saga.js]: We might need to think about the issue where the endpoint could be changed and the length of the string could be changed as well. Maybe we could use regex to extract the string in a safer way.

Comment (liamaharon) [reply src/containers/HubiiApiHoc/saga.js]: Very good point, will change to regex

Comment (katat) [reply src/containers/HubiiApiHoc/saga.js]: Or we could isolate that string into a new property in the `network` object, and convert `walletApiEndpoint` into a function that concats the host string property, so no need to do any extractions.

File: src/containers/HubiiApiHoc/saga.js [hunk 2]
```diff
@@ -98,28 +97,57 @@ export function* loadTransactions({ address }, endpoint) {
   }
 }
 
+export function* requestToken() {
+  while (true) { // eslint-disable-line no-constant-condition
```
Comment (katat) [inline src/containers/HubiiApiHoc/saga.js]: Just curious, what is the frequency the token would be changed?

Comment (liamaharon) [reply src/containers/HubiiApiHoc/saga.js]: 1 minute, there's a timer on ln 142

**Pull request merged**